numpy/random/mtrand/mtrand.pyx

```diff
@@ -574,6 +579,128 @@ def _shape_from_size(size, d):
            shape = tuple(size) + (d,)
     return shape
 
+
+# Set up dictionary of integer types and relevant functions.
+#
+# The dictionary is keyed by dtype(...).name and the values
+# are a tuple (low, high, function), where low and high are
+# the bounds of the largest half open interval `[low, high)`
+# and the function is the relevant function to call for
+# that precision.
+
+
+def rand_int32(low, high, size, rngstate):
+    """
+    random_int32(self, low, high=None, size=None, state)
+
+    Return random np.int32 integers between `low` and `high`, inclusive.
+
+    Return random integers from the "discrete uniform" distribution in the
+    closed interval [`low`, `high`).  If `high` is None (the default),
+    then results are from [0, `low`).
+
+    Parameters
+    ----------
+    low : int
+        Lowest (signed) integer to be drawn from the distribution (unless
+        ``high=None``, in which case this parameter is the *highest* such
+        integer).
+    high : int
+        If provided, the largest (signed) integer to be drawn from the
+        distribution (see above for behavior if ``high=None``).
+    size : int or tuple of ints
+        Output shape.  If the given shape is, e.g., ``(m, n, k)``, then
+        ``m * n * k`` samples are drawn.  Default is None, in which case a
+        single value is returned.
+
```

Comment:
Judging from the email discussion, it seems though people are more in favor to implement the lower bit count versions in ```randomkit.c``` and hence to avoid extra documentation as @charris mentioned I think in his PR description.

Nevertheless, if my comments can attest to anything, I do agree that the documentation does seem kind of shaky right now.  Though it is WIP, and since there is a lot of functionality left to implement (assuming that @charris intends to add ```dtype``` functionality in full), the documentation could probably be tightened up at the end in case anything changes along the way.